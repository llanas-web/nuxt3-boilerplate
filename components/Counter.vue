<template>
  <div
    class="flex flex-col items-center justify-center h-full space-y-4 text-xl"
  >
    <p>Here's a counter:</p>
    <div class="flex items-center justify-center">
      <button
        class="px-4 py-2 font-bold text-white bg-green-500 rounded hover:bg-green-700"
        @click="() => counter++"
      >
        +
      </button>
      <span class="mx-2">{{ counter }}</span>
      <button
        class="px-4 py-2 font-bold text-white bg-green-500 rounded hover:bg-green-700"
        @click="() => counter--"
      >
        -
      </button>
    </div>
  </div>
</template>

<script lang="ts" setup>
const counter = 0;
</script>
