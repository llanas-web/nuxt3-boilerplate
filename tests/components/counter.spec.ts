import Counter from "@/components/Counter.vue";
import { mount } from "@vue/test-utils";

describe("counter.vue", () => {
  it("should render correctly", () => {
    const wrapper = mount(Counter);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
