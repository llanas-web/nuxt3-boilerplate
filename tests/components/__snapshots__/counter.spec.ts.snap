// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter.vue should render correctly 1`] = `
"<div class=\\"flex flex-col items-center justify-center h-full space-y-4 text-xl\\">
  <p>Here's a counter:</p>
  <div class=\\"flex items-center justify-center\\"></div>
</div>"
`;
